src/examples/zkprogram/dynamic-keys-merkletree.ts

```diff
@@ -30,13 +31,27 @@ const sideloadedProgram = ZkProgram({
         return publicInput.add(privateInput);
       },
     },
+    assertAndAdd: {
+      privateInputs: [Field],
+      async method(publicInput: Field, privateInput: Field) {
+        // this uses assert to test range check gates and their feature flags
+        publicInput.assertLessThanOrEqual(privateInput);
+        return publicInput.add(privateInput);
+      },
+    },
   },
 });
 
+// given a zkProgram, we compute the feature flags that we need in order to verify proofs that were generated
+const featureFlags = await FeatureFlags.fromZkProgram(sideloadedProgram);
+
 class SideloadedProgramProof extends DynamicProof<Field, Field> {
   static publicInputType = Field;
   static publicOutputType = Field;
   static maxProofsVerified = 0 as const;
+
+  // we use the feature flags that we computed from the `sideloadedProgram` ZkProgram
+  static featureFlags = featureFlags;
```

Comment:
Why must we pass this in if we are computing the feature flags based on the `ZkProgram`? Why wouldn't we just generate at runtime after compiling?

Reply:
Good question! We have to know the feature flag configuration at compile time, in advance. 
We use this for side loading, which means we can, in theory, verify proofs from any zkProgram, no matter how exactly the circuit looks like. The ONLY thing that we need to know in advance are the feature flags! Hence we give the developer to either specify them manually, or compute them from _any_ zkProgram as sort of "blueprint", if that makes sense!

Reply:
Oh, and the ZkProgram the developer wants to use as a blueprint could be any!

Reply:
Makes sense to me! For some reason when I was reading this PR, I thought we were attaching this to the `ZkProgram` itself, not the side loaded proof!